src/main/java/org/java_websocket/WebSocketAdapter.java

```diff
@@ -40,6 +40,8 @@
  **/
 public abstract class WebSocketAdapter implements WebSocketListener {
 
+	private PingFrame pingFrame;
```

Comment:
Do you even need to cache it anymore? Should the user not be able to send each cycle specific data?

Reply:
I believe that the cache is used only for the default implementation. If the user overrides this method, the frame will not be cached.

Reply:
I think most people will not be overriding the default implementation, so a cache could be useful. Atleast that's what I thought, though I can change it if you like.

Reply:
Yes, missed it.